pkg/daemon/daemon.go

```diff
@@ -119,153 +148,194 @@ func (d *daemon) Run() {
+
+	if err := d.kubeClient.SetAnnotationsOnPod(pi.pod, pi.pod.Annotations); err != nil {
+		if kerrors.IsNotFound(err) {
+			return fmt.Errorf("failed to update pod annotations with err: %v", err)
+		}
+
+		if err = d.guidPool.ReleaseGUID(pi.addr.String()); err != nil {
+			log.Warn().Msgf("failed to release guid \"%s\" from removed pod \"%s\" in namespace "+
+				"\"%s\" with error: %v", pi.addr.String(), pi.pod.Name, pi.pod.Namespace, err)
+		} else {
+			delete(d.guidPodNetworkMap, pi.addr.String())
+		}
+
+		*removedList = append(*removedList, pi.addr)
+	}
+
+	return nil
+}
+
 func (d *daemon) AddPeriodicUpdate() {
 	log.Info().Msgf("running periodic add update")
 	addMap, _ := d.watcher.GetHandler().GetResults()
 	addMap.Lock()
 	defer addMap.Unlock()
-	podNetworksMap := map[types.UID][]*v1.NetworkSelectionElement{}
+	// Contains ALL pods' networks
+	netMap := networksMap{theMap: make(map[types.UID][]*v1.NetworkSelectionElement)}
 	for networkID, podsInterface := range addMap.Items {
-		log.Info().Msgf("processing network networkID %s", networkID)
-		networkNamespace, networkName, err := utils.ParseNetworkID(networkID)
-		if err != nil {
-			log.Err(err)
-			continue
-		}
 		pods, ok := podsInterface.([]*kapi.Pod)
 		if !ok {
-			log.Error().Msgf(
-				"invalid value for add map networks expected pods array \"[]*kubernetes.Pod\", found %T",
-				podsInterface)
+			log.Error().Msgf("invalid value for add map networks expected pods array \"[]*kubernetes.Pod\""+
+				", found %T", podsInterface)
 			continue
 		}
 
 		if len(pods) == 0 {
 			continue
 		}
 
-		netAttInfo, err := d.kubeClient.GetNetworkAttachmentDefinition(networkNamespace, networkName)
-		if err != nil {
-			log.Warn().Msgf("failed to get networkName attachment %s with error: %v", networkName, err)
-			// skip failed networks
-			continue
-		}
-
-		log.Debug().Msgf("networkName attachment %v", netAttInfo)
-		networkSpec := make(map[string]interface{})
-		err = json.Unmarshal([]byte(netAttInfo.Spec.Config), &networkSpec)
-		if err != nil {
-			log.Warn().Msgf("failed to parse networkName attachment %s with error: %v", networkName, err)
-			// skip failed networks
-			continue
-		}
-		log.Debug().Msgf("networkName attachment spec %+v", networkSpec)
-
-		ibCniSpec, err := utils.GetIbSriovCniFromNetwork(networkSpec)
+		log.Info().Msgf("processing network networkID %s", networkID)
+		networkName, ibCniSpec, err := d.getIbSriovNetwork(networkID)
 		if err != nil {
-			addMap.UnSafeRemove(networkID)
-			log.Warn().Msgf("failed to get InfiniBand SR-IOV CNI spec from network attachment %+v, with error %v",
-				networkSpec, err)
-			// skip failed network
+			if errors.ErrorCode(err) == errors.NotValidError {
+				// Not an IB SR-IOV network
+				addMap.UnSafeRemove(networkID)
+			}
+			log.Error().Msgf("%v", err)
 			continue
 		}
-		log.Debug().Msgf("CNI spec %+v", ibCniSpec)
 
 		var guidList []net.HardwareAddr
-		var passedPods []*kapi.Pod
+		var passedPodsInfo []*podInfo
```

Comment:
IMO you can keep this as was before, i.e `passedPods` or `passedPodInfos`  where the `s` at the end helps the reader remember its a list.

however if you want to keep the change im fine as well.